src/contracts/atlas/GasAccounting.sol

```diff
@@ -201,107 +208,187 @@ abstract contract GasAccounting is SafetyLocks {
-        // TODO: consider penalizing bundler for too much unused gas (to prevent high escrow requirements for solvers)
-        uint256 gasRemainder = gasleft() * tx.gasprice;
-        gasRemainder = gasRemainder * (SURCHARGE_SCALE + SURCHARGE_RATE) / SURCHARGE_SCALE;
-        _claims -= gasRemainder;
+        // Handle the settlement accounting logs
+        if (!ctx.solverSuccessful) {
+            // CASE: No solver was successful
 
-        if (_deposits < _claims + _withdrawals) {
-            // CASE: in deficit, subtract from bonded balance
-            uint256 amountOwed = _claims + _withdrawals - _deposits;
-            uint256 deficit = _assign(winningSolver, amountOwed, true, false);
+            // Ignore assigning to the winningSolver, just refund the bet deposits to the bundler if applicable
+            if (_deposits < _withdrawals + netAtlasGasSurcharge) {
+                // NOTE: We assume the bundler is fully on the hook for the gas, so we ignore claims and writeoffs
+                // and focus exclusively on deposits and withdrawals.
+                // The "+ netAtlasGasSurcharge" forces the bundler to pay the surcharge
+                revert InsufficientTotalBalance(_withdrawals + netAtlasGasSurcharge - _deposits);
+            }
+            claimsPaidToBundler = _deposits - _withdrawals;
+        } else if (_winningSolver == ctx.bundler) {
+            // CASE: The winning solver is also the bundler
+            if (_deposits < _withdrawals + netAtlasGasSurcharge) {
+                // CASE: in deficit, subtract from bonded balance
+                uint256 amountOwed = _withdrawals + netAtlasGasSurcharge - _deposits;
+                uint256 deficit = _assign(_winningSolver, amountOwed, true);
+                if (deficit > 0) {
+                    revert InsufficientTotalBalance(deficit); // Revert if insufficient bonded balance.
+                }
+                claimsPaidToBundler = 0; // Bundler-Solver is in deficit, no need to pay them.
+            } else {
+                // CASE: in surplus, add to bonded balance by crediting the bundler at bottom of function
+                claimsPaidToBundler = _deposits - _withdrawals - netAtlasGasSurcharge;
+            }
+        } else if (_writeoffs + _deposits < _claims + _withdrawals) {
+            // CASE: Not a special bundler, solver successful, balance in deficit.
+            // NOTE _claims and _writeoffs already have the Gas Surcharges factored in
+            uint256 amountOwed = _claims + _withdrawals - _writeoffs - _deposits;
+            uint256 deficit = _assign(_winningSolver, amountOwed, true);
             if (deficit > 0) {
-                revert InsufficientTotalBalance(deficit);
+                // CASE: Solver's bonded balance isn't enough to cover the amount owed, and the
+                // winning solver is unrelated to the bundler. The bundler is not the solver.
+                if (deficit > _claims - _writeoffs - netAtlasGasSurcharge) {
```

Comment:
you commented that "_claims and _writeoffs already have the Gas Surcharges factored in" above this